From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Niels Breet <niels.breet@jolla.com>
Date: Tue, 11 May 2021 13:10:49 +0300
Subject: [PATCH] Scratchbox2 needs to be able to tell rustc the default
 target.

Currently this defaults to the target rust was built with. Rather than
hardcode this for a cross-compiler, using an environment variable
allows the same rustc binary to be used for all builds.

Signed-off-by: David Greaves <david.greaves@jolla.com>
Signed-off-by: Ruben De Smet <ruben.de.smet@rubdos.be>
---
 compiler/rustc_session/src/config.rs | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/compiler/rustc_session/src/config.rs b/compiler/rustc_session/src/config.rs
index 4182a5d07118e0652ff63c4bec28bde0775d0c59..9904ab2e70d7aafab4ec63e86d0cdb383ca2285b 100644
--- a/compiler/rustc_session/src/config.rs
+++ b/compiler/rustc_session/src/config.rs
@@ -1848,7 +1848,14 @@ pub fn parse_target_triple(
             })
         }
         Some(target) => TargetTriple::TargetTriple(target),
-        _ => TargetTriple::from_triple(host_triple()),
+        // In SFOS we use SB2 and need to tell rust what the
+        // 'default' target is. If the SB2_RUST_TARGET_TRIPLE
+        // environment variable is set then it is used. Otherwise
+        // the fallback is the host_triple as usual
+        _ => match std::env::var("SB2_RUST_TARGET_TRIPLE") {
+            Ok(tgt) => TargetTriple::TargetTriple(tgt),
+            Err(_) => TargetTriple::from_triple(host_triple()),
+        }
     }
 }
 
